logbook-core/src/main/java/org/zalando/logbook/JsonHttpLogFormatter.java

```diff
@@ -100,7 +100,7 @@ private void addBody(final HttpMessage request, final ImmutableMap.Builder<Strin
         final String body = request.getBodyAsString();
 
         if (isJson(request.getContentType())) {
-            builder.put("body", new JsonBody(compactJson(body)));
+            builder.put("body", body.isEmpty() ? JsonBody.NULL : new JsonBody(compactJson(body)));
```

Comment:
Should this be an empty string instead? Otherwise you can't distinguish between an empty body and a body containing the string "null".